src/process/qgsprocess.cpp

```diff
@@ -0,0 +1,505 @@
+{
+  std::cout << info.toLocal8Bit().constData() << '\n';
+}
+
+void ConsoleFeedback::pushConsoleInfo( const QString &info )
+{
+  std::cout << info.toLocal8Bit().constData() << '\n';
+}
+
+void ConsoleFeedback::showTerminalProgress( double progress )
+{
+  const int thisTick = std::min( 40, std::max( 0, static_cast<int>( progress * 0.4 ) ) );
+
+  if ( mTimer.elapsed() > 2000 )
+  {
+    // check for signals every 2s to allow for responsive cancellation
+    QCoreApplication::processEvents();
+    mTimer.restart();
+  }
+
+  // Have we started a new progress run?
+  if ( thisTick <= mLastTick )
+    return;
+
+  while ( thisTick > mLastTick )
+  {
+    ++mLastTick;
+    if ( mLastTick % 4 == 0 )
+      fprintf( stdout, "%d", ( mLastTick / 4 ) * 10 );
+    else
+      fprintf( stdout, "." );
+  }
+
+  if ( thisTick == 40 )
+    fprintf( stdout, " - done.\n" );
+  else
+    fflush( stdout );
+}
+
+
+//! load Python support if possible
+std::unique_ptr< QgsPythonUtils > loadPythonSupport()
+{
+  QString pythonlibName( QStringLiteral( "qgispython" ) );
+#if defined(Q_OS_UNIX) and !defined(Q_OS_ANDROID)
+  pythonlibName.prepend( QgsApplication::libraryPath() );
+#endif
+#ifdef __MINGW32__
+  pythonlibName.prepend( "lib" );
+#endif
+  QString version = QStringLiteral( "%1.%2.%3" ).arg( Qgis::versionInt() / 10000 ).arg( Qgis::versionInt() / 100 % 100 ).arg( Qgis::versionInt() % 100 );
+  QgsDebugMsg( QStringLiteral( "load library %1 (%2)" ).arg( pythonlibName, version ) );
+  QLibrary pythonlib( pythonlibName, version );
+  // It's necessary to set these two load hints, otherwise Python library won't work correctly
+  // see http://lists.kde.org/?l=pykde&m=117190116820758&w=2
+  pythonlib.setLoadHints( QLibrary::ResolveAllSymbolsHint | QLibrary::ExportExternalSymbolsHint );
+  if ( !pythonlib.load() )
+  {
+    pythonlib.setFileName( pythonlibName );
+    if ( !pythonlib.load() )
+    {
+      std::cerr << QStringLiteral( "Couldn't load Python support library: %1" ).arg( pythonlib.errorString() ).toLocal8Bit().constData();
+      return nullptr;
+    }
+  }
+
+  typedef QgsPythonUtils*( *inst )();
+  inst pythonlib_inst = reinterpret_cast< inst >( cast_to_fptr( pythonlib.resolve( "instance" ) ) );
+  if ( !pythonlib_inst )
+  {
+    //using stderr on purpose because we want end users to see this [TS]
+    std::cerr << "Couldn't resolve python support library's instance() symbol.";
+    return nullptr;
+  }
+
+  std::unique_ptr< QgsPythonUtils > pythonUtils( pythonlib_inst() );
+  if ( pythonUtils )
+  {
+    pythonUtils->initPython( nullptr, false );
+  }
+
+  return pythonUtils;
+}
+
+QgsProcessingExec::QgsProcessingExec()
+{
+
+}
+
+int QgsProcessingExec::run( const QStringList &args )
+{
+  if ( args.size() == 1 )
+  {
+    showUsage( args.at( 0 ) );
+    return 0;
+  }
+
+  // core providers
+  QgsApplication::processingRegistry()->addProvider( new QgsNativeAlgorithms( QgsApplication::processingRegistry() ) );
```

Comment:
sorry to arrive late... what about the other available/installed providers?

Reply:
I mean, I can see that you load all processing plugin... do we already moved gdal, saga and grass as in plugins area?

Reply:
It should be done in https://github.com/qgis/QGIS/pull/34617/files/1e55212fd563b532fca57d8e0a8ce95cf40f9f27#diff-d8e4bd8d4ecd2dd30113b5c3a6cd2a1aR254 -- but looking at this now, I think we need to ensure that the main python processing plugin (where the gdal/grass/saga/otb initialisation occurs) is loaded before any other plugin. I'll push a follow up